src/yunohost/alias.py

```diff
@@ -0,0 +1,188 @@
+
+    result = auth.search('ou=aliases,dc=yunohost,dc=org', filter, attrs)
+
+    if len(result) > offset and limit > 0:
+        for alias in result[offset:offset+limit]:
+            entry = {}
+            for attr, values in alias.items():
+                try:
+                    entry[alias_attrs[attr]] = values[0:]
+                except:
+                    pass
+            result_list.append(entry)
+    return { 'alias' : result_list }
+
+
+def alias_create(auth, alias, mailforward):
+    """
+    Create alias
+
+    Keyword argument:
+        alias -- Main mail address must be unique
+        mailforward -- List of email to forward, separated by commas without space
+
+    """
+    from yunohost.domain import domain_list
+
+    _ensure_ou_created(auth)
+
+    # Validate uniqueness of alias and mail in LDAP
+    auth.validate_uniqueness({
+        'mail'      : alias
+    })
+
+    # Check that the mail domain exists
+    if alias[alias.find('@')+1:] not in domain_list(auth)['domains']:
+        raise MoulinetteError(errno.EINVAL,
+                              m18n.n('mail_domain_unknown',
+                                     alias[alias.find('@')+1:]))
+
+    # Adapt values for LDAP
+    rdn = 'mail=%s,ou=aliases' % alias
+    attr_dict = {
+        'objectClass'   : ['mailAccount', 'mailAlias'],
+        'mail'          : alias
+    }
+
+    attr_dict['maildrop'] = mailforward.split(",")
+
+    if auth.add(rdn, attr_dict):
+        msignals.display(m18n.n('alias_created'), 'success')
+        return { 'alias' : alias, 'mailforward' : attr_dict['maildrop'] }
+
+    raise MoulinetteError(169, m18n.n('alias_creation_failed'))
+
+
+def alias_delete(auth, alias):
+    """
+    Delete alias
+
+    Keyword argument:
+        alias -- Alias to delete
+
+    """
+    _ensure_ou_created(auth)
+
+    if auth.remove('mail=%s,ou=aliases' % alias):
+        pass
+    else:
+        raise MoulinetteError(169, m18n.n('alias_deletion_failed'))
+
+    msignals.display(m18n.n('alias_deleted'), 'success')
+
+
+def alias_info(auth, alias):
+    """
+    Get alias informations
+
+    Keyword argument:
+        alias -- Alias mail to get informations
+
+    """
+    _ensure_ou_created(auth)
+
+    alias_attrs = [
+        'mail', 'maildrop'
+    ]
+
+    if len(alias.split('@')) is 2:
+        filter = 'mail=' + alias
+    else:
+        # TODO better error message
+        raise MoulinetteError(167, m18n.n('alias_info_failed'))
+
+    result = auth.search('ou=aliases,dc=yunohost,dc=org', filter, alias_attrs)
+
+    if result:
+        alias = result[0]
+    else:
+        raise MoulinetteError(errno.EINVAL, m18n.n('alias_unknown'))
+
+    result_dict = {
+        'alias': alias['mail'][0]
+    }
+
+    if len(alias['maildrop']) > 0:
+        result_dict['mail-forward'] = alias['maildrop'][0:]
+
+    if result:
+        return result_dict
+    else:
+        raise MoulinetteError(167, m18n.n('alias_info_failed'))
+
+
+def _ensure_ou_created(auth):
```

Comment:
Would be more obvious if named something like `_ensure_ldap_ou_is_created`.

That would be a good candidate for the `migration` PR to avoid having to check every time.